File: ocaml/consultar.ml
(*
    Exemplo de uma chamada na API da base na linguagem OCaml
    author Joubert <me+github@redrat.com.br>
*)
open Cohttp_lwt_unix
open Cohttp
open Lwt

let uri = Uri.of_string "http://subdominio.dominio.exemplo/api/v1/marcas/'" in
let headers = Header.add_list (Header.init ()) [
  ("x-auth-token", "33db934c-7f44-4dd0-b8e3-955af5e3643a");
] in

Client.call ~headers `GET uri
>>= fun (res, body_stream) ->
  (* Do stuff with the result *)

File: node-js/consultar.js
/*
 * Exemplo de uma chamada na API da base na linguagem Javascript com node.js
 *
 * author Joubert <me+github@redrat.com.br>
 */

var http = require("http");

var options = {
  "method": "GET",
  "hostname": "subdominio.dominio.exemplo",
  "port": null,
  "path": "/api/v1/marcas/",
  "headers": {
    "x-auth-token": "33db934c-7f44-4dd0-b8e3-955af5e3643a"
  }
};

var req = http.request(options, function (res) {
  var chunks = [];

  res.on("data", function (chunk) {
    chunks.push(chunk);
  });

  res.on("end", function () {
    var body = Buffer.concat(chunks);
    console.log(body.toString());
  });
});

req.end();
